src/main/java/dukebot/Duke.java

```diff
@@ -0,0 +1,134 @@
+package dukebot;
+
+import dukebot.command.Command;
+import dukebot.command.Parser;
+import dukebot.contactlist.ContactDetails;
+import dukebot.contactlist.ContactList;
+import dukebot.exception.DukeException;
+import dukebot.storage.AppStorage;
+import dukebot.storage.Storage;
+import dukebot.tasklist.Task;
+import dukebot.tasklist.TaskList;
+import dukebot.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Main logic.
+ */
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private final Parser parser;
+    private AppStorage appStorage;
+
+    /**
+     * Initialises Duke.
+     *
+     * @param withGui Whether GUI mode is used.
+     */
+    public Duke(boolean withGui) {
+        storage = new Storage();
+        ui = new Ui(withGui);
+
+        // Load Aliases
+        Parser parserTemp;
+        try {
+            parserTemp = new Parser(storage.loadAlias());
+        } catch (DukeException e) {
+            // Can ignore the exception message
```

Comment:
The comment may be unnecessary if you can still understand your code without it :)

Reply:
I feel that there should always be some explanation if the error message isn't handled for future reference.